app/models/planting.rb

```diff
@@ -8,6 +8,8 @@ class Planting < ActiveRecord::Base
   belongs_to :crop, counter_cache: true
   has_many :harvests, -> { order(harvested_at: :desc) }, dependent: :destroy
 
+  before_save :calc_and_set_days_before_maturity
```

Comment:
Not a huge fan of before_saves, could this be better done with a planting updater pattern? IE any controller type action that would update maturity uses that; while other simple saves just use .save (rake tasks for example)